Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -898,6 +904,14 @@ class JSGlobalObject : public JSSegmentedVariableObject {
     {
         return originalArrayStructureForIndexingType(structure->indexingMode() | IsArray) == structure;
     }
+
+    bool isOriginalSlowPutContigiousStructure(Structure* structure)
+    {
+        ASSERT(hasAlwaysSlowPutContiguous(structure->indexingMode()));
+        return m_originalAlwaysSlowPutContiguousStructures.contains(structure->id());
+    }
```

Comment:
Instead of this, how about checking `didTransition` bit of this structure? Does that work?

Reply:
This bit was re-written into having `JSGlobalObject::m_originalAlwaysSlowPutContiguousStructureSet`, which we either way need to emit `CheckStructure`.

Reply:
`didTransition` + `indexingMode` check would indeed worked out there.